components/lib/jexl/src/main/java/mozilla/components/lib/jexl/parser/StateMachine.kt

```diff
@@ -141,7 +144,10 @@ internal val stateMachine: Map<State, StateMap> = mapOf(
     ),
     State.ARGUMENT_VALUE to StateMap(
         subHandler = { parser, node ->
-            parser.cursor!!.arguments.add(node!!)
+            val cursor = parser.cursor!!
+            if (cursor is Transformation) {
+                cursor.arguments.add(node!!)
+            }
```

Comment:
@pocmo Maybe I should throw exception here if node is incorrect? What do you think?

Reply:
Or even make cast that will throw exception.

Reply:
Yes, that sounds good - especially if this keeps the tests green. :)

Reply:
:heavy_check_mark: